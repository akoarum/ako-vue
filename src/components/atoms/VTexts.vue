<template>
  <p :class="styledClasses" class="vTexts"><slot /></p>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    caution: { type: Boolean },
    small: { type: Boolean }
  },
  computed: {
    styledClasses(): string[] {
      const classes: string[] = []
      if (this.caution) classes.push('-caution')
      if (this.small) classes.push('-small')
      return classes
    }
  }
})
</script>

<style lang="scss" scoped>
.vTexts {
  white-space: pre-line;

  @include media-min {
    font-size: $fontSize-pc;
    line-height: 1.8;
  }

  @include media-max {
    font-size: $fontSize-sp;
    line-height: 1.7;
  }
}

.-caution {
  color: $color-caution;
}

.-small {
  @include media-min {
    font-size: 1.4rem;
    line-height: 1.5;
  }

  @include media-max {
    font-size: 1.3rem;
    line-height: 1.4;
  }
}
</style>
