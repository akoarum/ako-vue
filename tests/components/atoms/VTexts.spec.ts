import { shallowMount } from '@vue/test-utils'
import VTexts from '~/components/atoms/VTexts.vue'

describe('VTexts', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VTexts, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClasses: propsを基にクラス配列を生成して返す', () => {
      expect(wrapper.vm.styledClasses).toEqual([])

      wrapper.setProps({ caution: true })
      expect(wrapper.vm.styledClasses).toEqual(['-caution'])

      wrapper.setProps({ small: true })
      expect(wrapper.vm.styledClasses).toEqual(['-caution', '-small'])
    })
  })
})
