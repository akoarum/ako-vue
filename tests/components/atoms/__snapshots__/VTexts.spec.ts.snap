// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTexts initialized snapshot 1`] = `
<p
  class="vTexts"
/>
`;
